hazelcast/src/test/java/com/hazelcast/replicatedmap/ReplicatedMapStatsTest.java

```diff
@@ -0,0 +1,251 @@
+        assertEquals(100, stats.getHits());
+    }
+
+    @Test
+    public void testPutAndHitsGenerated() throws Exception {
+        ReplicatedMap<Integer, Integer> replicatedMap = getReplicatedMap();
+        for (int i = 0; i < 100; i++) {
+            replicatedMap.put(i, i);
+            replicatedMap.get(i);
+        }
+        LocalReplicatedMapStats stats = replicatedMap.getReplicatedMapStats();
+        assertEquals(100, stats.getHits());
+    }
+
+    @Test
+    public void testGetAndHitsGenerated() throws Exception {
+        ReplicatedMap<Integer, Integer> replicatedMap = getReplicatedMap();
+        for (int i = 0; i < 100; i++) {
+            replicatedMap.put(i, i);
+            replicatedMap.get(i);
+        }
+        LocalReplicatedMapStats stats = replicatedMap.getReplicatedMapStats();
+        assertEquals(100, stats.getHits());
+    }
+
+    @Test
+    public void testHitsGenerated_updatedConcurrently() throws Exception {
+        final ReplicatedMap<Integer, Integer> replicatedMap = getReplicatedMap();
+        final int actionCount = 100;
+        for (int i = 0; i < actionCount; i++) {
+            replicatedMap.put(i, i);
+            replicatedMap.get(i);
+        }
+        final LocalReplicatedMapStats stats = replicatedMap.getReplicatedMapStats();
+        final long initialHits = stats.getHits();
+
+        new Thread(new Runnable() {
+            @Override
+            public void run() {
+                for (int i = 0; i < actionCount; i++) {
+                    replicatedMap.get(i);
+                }
+                replicatedMap.getReplicatedMapStats(); // causes the local stats object to update
+            }
+        }).start();
+
+        assertEquals(actionCount, initialHits);
+        assertTrueEventually(new AssertTask() {
+            @Override
+            public void run()
+                    throws Exception {
+                assertEquals(actionCount * 2, stats.getHits());
+            }
+        });
+    }
+
+    @Test
+    public void testLastAccessTime() throws InterruptedException {
+        final long startTime = Clock.currentTimeMillis();
+        ReplicatedMap<String, String> replicatedMap = getReplicatedMap();
+        String key = "key";
+        replicatedMap.put(key, "value");
+        replicatedMap.get(key);
+        long lastAccessTime = replicatedMap.getReplicatedMapStats().getLastAccessTime();
+
+        assertTrue(lastAccessTime >= startTime);
+    }
+
+    @Test
+    public void testLastAccessTime_updatedConcurrently() throws InterruptedException {
+        final long startTime = Clock.currentTimeMillis();
+        final ReplicatedMap<String, String> map = getReplicatedMap();
+        final String key = "key";
+        map.put(key, "value");
+        map.get(key);
+        final LocalReplicatedMapStats stats = map.getReplicatedMapStats();
+        final long lastAccessTime = stats.getLastAccessTime();
+
+        new Thread(new Runnable() {
```

Comment:
There is a HazelcastTestSupport.spawn method that can make the code a bit cleaner by removing the thread/start. Not that important.